usm_rest_api/v1/views/tasks.py

```diff
@@ -0,0 +1,224 @@
+                     (cluster_data, nodelist))
+        return {'status': 'ALL_NODES_FAILURE', 'failednodes': str(nodelist)}
+    else:
+        # merge the failed list
+        failedNodes += failed
+    log.debug("Done. Executed the commands on nodes")
+    successNodes = [item for item in probeList if item not in failedNodes]
+    log.debug("Successful Nodes %s" % successNodes)
+    log.debug("Failed Nodes %s" % failedNodes)
+    current_task.update_state(state='UPDATE_CLUSTER_STATUS')
+    try:
+        #
+        # Update the status of the successful Nodes in DB
+        # HOST_STATUS_INACTIVE = 1
+        # HOST_STATUS_ACTIVE = 2
+        #
+        usm_rest_utils.update_host_status(
+            successNodes, usm_rest_utils.HOST_STATUS_ACTIVE)
+        # Update the status of Cluster to Active if atlest two nodes
+        # are added successfully for gluster cluster
+        # Update the status of Cluster to Active if atlest one nodes
+        # are added successfully for ceph cluster
+        # STATUS_INACTIVE = 1
+        # STATUS_ACTIVE_NOT_AVAILABLE = 2
+        # STATUS_ACTIVE_AND_AVAILABLE = 3
+        #
+        if (len(successNodes) >= 1 and cluster_data['cluster_type']
+                == usm_rest_utils.CLUSTER_TYPE_CEPH) or \
+                (len(successNodes) >= 2 and cluster_data['cluster_type'] ==
+                 usm_rest_utils.CLUSTER_TYPE_GLUSTER):
+            usm_rest_utils.update_cluster_status(
+                str(cluster_data['cluster_id']),
+                usm_rest_utils.STATUS_ACTIVE_AND_AVAILABLE)
+
+    except Exception, e:
+        log.exception(e)
+
+    if noOfNodes == len(failedNodes):
+        log.critical("Cluster creation Failed %s and Nodelist %s" %
+                     (cluster_data, nodelist))
+        return {'status': 'ALL_NODES_FAILURE', 'failednodes': str(failedNodes)}
+    elif failedNodes:
+        log.critical("Cluster creation partially Failed %s and Nodelist %s" %
+                     (cluster_data, failedNodes))
+        return {'status': 'PARTIAL_FAILURE', 'failednodes': str(failedNodes)}
+
+    return states.SUCCESS
+
+
+@shared_task
+def createHost(data):
+    log.debug("Inside createGlusterHost Async Task %s" % data)
+    cluster = None
+    cluster_data = None
+    hostlist = None
+
+    # Get the cluster configuration
+    try:
+        # Get the cluster config
+        cluster = Cluster.objects.get(pk=str(data['cluster']))
+        cluster_data = {"cluster_id": cluster.cluster_id,
+                        "cluster_name": cluster.cluster_name,
+                        "cluster_type": cluster.cluster_type,
+                        "storage_type": cluster.storage_type}
+        log.debug("Cluster Data %s" % cluster_data)
+        # Get nodes belongs to the cluster for peerprobe
+        hostlist = Host.objects.filter(cluster_id=str(data['cluster']))
+        log.debug("Hosts in the cluster: %s" % hostlist)
+    except Exception, e:
```

Comment:
I will leave this as it is for the time being. Fix this when backend code returns proper exceptions